haystack/preview/components/retrievers/memory_bm25_retriever.py

```diff
@@ -41,6 +41,12 @@ def __init__(
         self.top_k = top_k
         self.scale_score = scale_score
 
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"document_store": type(self.document_store).__name__}
```

Comment:
This is to see how often this retriever is wrongly used with another document store or to account for any future retriever inheriting from MemoryBM25Retriever?

Reply:
Same question for MemoryEmbeddingRetriever and DocumentWriter 🙂

Reply:
I think it's better if all components that use Document Store to send the type of docstore they use, regardless whether they're a specialized class like MemoryBM25Retriever or not, because that helps aggregating the values in PostHog side. 

For Document Writer it especially important because it can use any docstore.